test/conf_include_test.c

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2016-2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <openssl/conf.h>
+#include <openssl/err.h>
+#include "testutil.h"
+
+#ifdef _WIN32
+# include <direct.h>
+# define DIRSEP "/\\"
+# define chdir _chdir
+# define DIRSEP_PRESERVE 0
+#elif !defined(OPENSSL_NO_POSIX_IO)
+# include <unistd.h>
+# ifndef OPENSSL_SYS_VMS
+#  define DIRSEP "/"
+#  define DIRSEP_PRESERVE 0
+# else
+#  define DIRSEP "/]:"
+#  define DIRSEP_PRESERVE 1
```

Comment:
On side note. [And side note means that no action is expected.] Given the exchange with Richard, it feels that DIRSEP_PRESERVE is required only if separator is `]` or `:`, but not `/`. I mean remark was that VMS chdir is of course VMS, but it understands Unix-y slashes too. And question is if the latter implies that it should get same treatment, i.e. last slash being replaced with `\0` and not character after slash. Well, on Unix it's not actually a requirement in sense that you can change directory to `/some/where` and `/some/where/` with equal success. And there is no reason to believe that VMS would be different, but it makes you wonder...

Reply:
I knew about that "glitch" but did not want to complicate the code more.